ntoskrnl/mm/section.c

```diff
@@ -3641,8 +3641,43 @@ MiRosUnmapViewOfSection(IN PEPROCESS Process,
                     BaseAddress, Process, Status);
             ASSERT(NT_SUCCESS(Status));
         }
-    }
 
+        /* Flush the segment to disk, if needed */
```

Comment:
We are in the `if (MemoryArea->VadNode.u.VadFlags.VadType != VadImageMap)` case: what kind of type would it be then? How does the VadImageMap type differs regarding unmapping/flushing than the other ones?

Reply:
That would be a data file mapping, or a physical memory mapping. You don't flush an image mapping to disk. Never. That would mean that you can modify data in e.g. a DLL when running code from it and screw the next program trying to run it.